middle_end/flambda2/from_lambda/lambda_to_flambda.ml

```diff
@@ -1043,12 +1044,18 @@ let rec cps acc env ccenv (lam : L.lambda) (k : cps_continuation)
         compile_staticfail acc env ccenv ~continuation
           ~args:(List.flatten args @ extra_args))
       k_exn
-  | Lstaticcatch (body, (static_exn, args), handler, layout) ->
+  | Lstaticcatch (body, (static_exn, args), handler, r, layout) ->
     maybe_insert_let_cont "staticcatch_result" layout k acc env ccenv
       (fun acc env ccenv k ->
+        let pop_region =
+          match r with
+          | Popped_region -> true
```

Comment:
So in the `pop_region` case, it looks like the difference with what would happen if we had `Lstaticcatch (_, _, Lexclave _, _)` is that we don't add an `End_region`, isn't that a problem?

Reply:
No, that's the point: the region will already have been popped when we enter the handler. Perhaps this would all be clearer if I named various things `parent_region` rather than `pop_region`?